File: src/php/PhantasyIslandPhpRemoteControl/radio/DebugRadio.php
<?php

require_once __DIR__ . '/RadioManager.php';
require_once __DIR__ . '/ApiModule.php';

use PhantasyIslandPhpRemoteControl\Radio\RadioManager;
use function PhantasyIslandPhpRemoteControl\Radio\as_sync;

$rm = new RadioManager();
$rm->connect();

echo $rm->create_msg_timestamp_id() . PHP_EOL;

echo json_encode($rm->ping()) . PHP_EOL;

// sync 模式（默认）
$data = as_sync($rm->radioApi->listRadioLocalObjectsIds());
print_r($data);

echo 'isSceneInit: ' . ($rm->radioApi->isSceneInit() ? 'true' : 'false') . PHP_EOL;
echo 'isRadioReachabilityCheckerInit: ' . ($rm->radioApi->isRadioReachabilityCheckerInit() ? 'true' : 'false') . PHP_EOL;
print_r($rm->radioApi->getAllRadioMaterial());
print_r($rm->radioApi->localRadioMaterial());
print_r($rm->radioApi->listRadioLocalObjectsIds());

File: src/rs/radio/debug_radio.rs
use crate::radio::radio_manager::RadioManager;
use crate::radio::api_module::SendResult;
use std::time::Duration;

pub fn debug_radio() {
    let rm = RadioManager::new();
    // rm.connect("http://127.0.0.1:60002"); // 这是一个示例，实际中可能需要处理 Arc

    println!("{}", rm.create_msg_timestamp_id());

    match rm.ping() {
        Some(res) => println!("Ping result: {:?}", res),
        None => println!("Ping timeout"),
    }

    match rm.radio_api.list_radio_local_objects_ids() {
        SendResult::Sync(Some(data)) => println!("Local objects: {:?}", data),
        _ => println!("Failed to list objects"),
    }

    println!("is_scene_init: {:?}", rm.radio_api.is_scene_init());
}
